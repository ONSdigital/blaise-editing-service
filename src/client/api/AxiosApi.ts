import axios, { AxiosRequestConfig } from 'axios';
import { AuthManager } from 'blaise-login-react-client';
import notFound from '../../server/helpers/axiosHelper';

function axiosConfig(): AxiosRequestConfig {
  const authManager = new AuthManager();
  return {
    headers: {
      'Content-Type': 'application/json',
      ...authManager.authHeader(),
    },
  };
}

export async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url, axiosConfig());

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export async function patchDataToNode(url: string, payload: any, notFoundError: string): Promise<void> {
  try {
    await axios.patch(url, payload, axiosConfig());
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}
