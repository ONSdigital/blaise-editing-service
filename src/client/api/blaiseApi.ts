import axios from 'axios';
import { Questionnaire } from 'blaise-api-node-client';
import { CaseDetails, CaseFactsheet } from '../../common/interfaces/case.interface';
import notFound from '../../common/helpers/axios.helper';

export async function getQuestionnaires(): Promise<Questionnaire[]> {
  try {
    const response = await axios.get('/api/questionnaires');

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('Unable to find questionnaires, please contact Richmond Rice');
    }
    throw new Error('Unable to retrieve questionnaires, please try again in a few minutes');
  }
}

export async function getCases(questionnaireName: string): Promise<CaseDetails[]> {
  try {
    const response = await axios.get(`/api/questionnaires/${questionnaireName}/cases`);
    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('The questionnaire is no longer available');
    }
    throw new Error('Unable to retrieve cases, please try again in a few minutes');
  }
}

export async function getCaseFactsheet(questionnaireName: string, caseId: string): Promise<CaseFactsheet> {
  try {
    const response = await axios.get(`/api/questionnaires/${questionnaireName}/cases/${caseId}/factsheet`);
    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error('The questionnaire is no longer available');
    }
    throw new Error('Unable to retrieve case factsheet, please try again in a few minutes');
  }
}
