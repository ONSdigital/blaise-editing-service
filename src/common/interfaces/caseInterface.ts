export interface CaseSummaryDetails {
  CaseId: string
  OutcomeCode: number,
  InterviewDate: Date,
  InterviewerName: string,
  InterviewerNumber: string,
  NumberOfRespondents: number,
  Household: {
    Address: {
      AddressLine1: string,
      AddressLine2: string,
      AddressLine3: string,
      AddressLine4: string,
      County: string,
      Town: string,
      Postcode: string,
    },
    Type: string,
    FloorNumber: number,
    Status: string,
    NumberOfBedrooms: number,
    ReceiptOfHousingBenefit: number,
    PeriodCode: number,
    CouncilTaxBand: string,
    BusinessRoom: boolean,
    SelfEmployed: boolean,
    SelfEmployedMembers: string,
    IncomeSupport: boolean,
    IncomeSupportMembers: string,
    IncomeBasesJaSupport: boolean,
    IncomeBasesJaSupportMembers: string,

  },
  Respondents: {
    RespondentName: string,
    DateOfBirth: Date,
  }[]
}
