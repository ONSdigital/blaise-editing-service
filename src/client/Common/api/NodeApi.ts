import axios from 'axios';
import notFound from '../../../server/helpers/axiosHelper';
import { Survey } from '../../../common/interfaces/surveyInterface';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export default async function getSurveys(): Promise<Survey[]> {
  return getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
}
