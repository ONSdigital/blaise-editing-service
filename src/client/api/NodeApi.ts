import axios from 'axios';

import { CaseEditInformation } from 'blaise-api-node-client/lib/cjs/interfaces/case';
import { User } from 'blaise-api-node-client/lib/cjs/interfaces/user';
import notFound from '../../server/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';
import { SupervisorInformation } from '../Interfaces/supervisorInterface';
import { EditorInformation } from '../Interfaces/editorInterface';
import mapEditorInformation from '../Mappers/editorInformaitionMapper';
import mapSupervisorInformation from '../Mappers/supervisorInformationMapper';
import { CaseSummaryDetails } from '../../common/interfaces/caseInterface';
import { CasesNotAllocatedInformation } from '../Interfaces/caseAllocationInterface';
import mapCasesNotAllocated from '../Mappers/caseAllocationMapper';
// import { caseSummaryDetailsMockObject } from '../../test/server/mockObjects/CaseMockObject';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export async function getSurveys(userRole: string): Promise<Survey[]> {
  return getDataFromNode(`/api/surveys?userRole=${userRole}`, 'Unable to find surveys, please contact Richmond Rice');
}

export async function getCaseSummary(questionnaireName: string, caseId: string): Promise<CaseSummaryDetails> {
  return getDataFromNode(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/${caseId}/summary`, 'The questionnaire is no longer available');
}

async function getCaseEditInformation(questionnaireName: string, userRole: string) {
  // TODO Fix the URL upper
  return getDataFromNode<CaseEditInformation[]>(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/edit?userRole=${userRole}`, 'Unable to find case edit information, please contact Richmond Rice');
}

export async function getEditorInformation(questionnaireName: string, editorUsername:string, editorRole: string): Promise<EditorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, editorRole);
  const caseEditInformationListForEditor = caseEditInformationList.filter((caseEditInformation) => caseEditInformation.assignedTo === editorUsername);

  return mapEditorInformation(caseEditInformationListForEditor);
}

export async function getSupervisorEditorInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<SupervisorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapSupervisorInformation(caseEditInformationList, editors);
}

export async function getCasesNotAllocatedInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<CasesNotAllocatedInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapCasesNotAllocated(caseEditInformationList, editors);
}
