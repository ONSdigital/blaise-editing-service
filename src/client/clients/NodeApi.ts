import axios from 'axios';
import { CaseDetails, CaseFactsheetDetails } from '../../common/interfaces/caseInterface';
import notFound from '../../common/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';

/* eslint-disable class-methods-use-this */
export default class NodeApi {
  constructor() {
    this.getSurveys = this.getSurveys.bind(this);
    this.getCases = this.getCases.bind(this);
    this.getCaseFactsheet = this.getCaseFactsheet.bind(this);
  }

  async getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
    try {
      const response = await axios.get(url);

      return response.data;
    } catch (error) {
      if (notFound(error)) {
        throw new Error(notFoundError);
      }
      throw new Error('Unable to complete request, please try again in a few minutes');
    }
  }

  async getSurveys(): Promise<Survey[]> {
    return this.getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
  }

  async getCases(questionnaireName: string): Promise<CaseDetails[]> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases`, 'The questionnaire is no longer available');
  }

  async getCaseFactsheet(questionnaireName: string, caseId: string): Promise<CaseFactsheetDetails> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases/${caseId}/factsheet`, 'The questionnaire is no longer available');
  }
}
