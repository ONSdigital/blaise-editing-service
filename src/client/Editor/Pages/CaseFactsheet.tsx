import { useParams } from 'react-router-dom';
import { useAsyncRequestWithTwoParams } from '../../Common/hooks/useAsyncRequest';
import { CaseFactsheetDetails } from '../../../common/interfaces/caseInterface';
import AsyncContent from '../../Common/components/AsyncContent';
import FactsheetContent from '../Components/FactsheetContent';
import { getCaseFactsheet } from '../../Common/api/NodeApi';

function DisplayCaseFactsheet(questionnaireName: string, caseId: string) {
  const caseFactsheet = useAsyncRequestWithTwoParams<CaseFactsheetDetails, string, string>(getCaseFactsheet, questionnaireName, caseId);

  return (
    <div data-testid="Factsheet">
      <AsyncContent content={caseFactsheet}>
        {(factsheetContent) => <FactsheetContent factsheet={factsheetContent} />}
      </AsyncContent>
    </div>
  );
}

export type CaseFactSheetParams = {
  questionnaireName: string
  caseId: string
};

export default function CaseFactSheet() {
  const { questionnaireName, caseId } = useParams<keyof CaseFactSheetParams>() as CaseFactSheetParams;

  return DisplayCaseFactsheet(questionnaireName, caseId);
}
