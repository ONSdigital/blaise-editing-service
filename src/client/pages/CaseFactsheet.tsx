import { useParams } from 'react-router-dom';
import { useAsyncRequestWithTwoParams } from '../hooks/useAsyncRequest';
import { CaseFactsheetDetails } from '../../common/interfaces/caseInterface';
import { getCaseFactsheet } from '../clients/serverApi';
import AsyncContent from '../components/AsyncContent';
import FactsheetContent from '../components/FactsheetContent';

function DisplayCaseFactsheet(questionnaireName: string, caseId: string) {
  const caseFactsheet = useAsyncRequestWithTwoParams<CaseFactsheetDetails>(getCaseFactsheet, questionnaireName, caseId);

  return (
    <AsyncContent content={caseFactsheet}>
      {(factsheetContent) => <FactsheetContent factsheet={factsheetContent} />}
    </AsyncContent>
  );
}

export default function CaseFactSheet() {
  const { questionnaireName, caseId } = useParams();
  if (!questionnaireName || !caseId) {
    return (
      <div>
        questionnaire name or caseId not provided
      </div>
    );
  }

  return DisplayCaseFactsheet(questionnaireName, caseId);
}
