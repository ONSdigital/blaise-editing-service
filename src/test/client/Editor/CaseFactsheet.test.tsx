import { RenderResult, act, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Router from 'react-router';
import { CaseFactsheetDetails } from '../../../common/interfaces/caseInterface';
import CaseFactsheet from '../../../client/Editor/Pages/CaseFactsheet';
import { caseFactsheetMockObject } from '../../mockObjects/caseMockObject';
import { getCaseFactsheet } from '../../../client/Common/api/NodeApi';

// declare global vars
const questionnaireName: string = 'TEST111A';
const caseId: string = '1';
let view:RenderResult;

// declare mocks
/* eslint import/no-extraneous-dependencies: 0 */
jest.mock('react-router', () => ({ ...jest.requireActual('react-router'), useParams: jest.fn() }));
jest.spyOn(Router, 'useParams').mockReturnValue({ questionnaireName, caseId });
jest.mock('../../../client/Common/api/NodeApi');
const getCaseFactsheetMock = getCaseFactsheet as jest.Mock<Promise<CaseFactsheetDetails>>;

describe('Given there is a case available in blaise for a questionnaire', () => {
  afterEach(() => {
    getCaseFactsheetMock.mockReset();
  });

  it('should render the factsheet page for the case correctly', async () => {
    // arrange
    const expectedCaseFactsheet: CaseFactsheetDetails = caseFactsheetMockObject;

    getCaseFactsheetMock.mockImplementation(() => Promise.resolve(expectedCaseFactsheet));

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseFactsheet />
        </BrowserRouter>,
      );
    });

    // assert
    const caseFactSheetView = view.getByTestId('Factsheet');
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.CaseId);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.AddressLine1);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.AddressLine2);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.AddressLine3);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.AddressLine4);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.County);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.Postcode);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.Address.Town);
    expect(caseFactSheetView).toHaveTextContent(expectedCaseFactsheet.InterviewerName);
    expect(caseFactSheetView).toHaveTextContent(String(expectedCaseFactsheet.NumberOfRespondents));
    expect(caseFactSheetView).toHaveTextContent(String(expectedCaseFactsheet.OutcomeCode));
    expectedCaseFactsheet.Respondents.forEach((respondent) => {
      expect(caseFactSheetView).toHaveTextContent(respondent.RespondentName);
      expect(caseFactSheetView).toHaveTextContent(String(respondent.DateOfBirth));
    });
  });

  it('should display the factsheet correctly', async () => {
    // arrange
    const expectedCaseFactsheet: CaseFactsheetDetails = caseFactsheetMockObject;

    getCaseFactsheetMock.mockImplementation(() => Promise.resolve(expectedCaseFactsheet));

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseFactsheet />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });
});

describe('Given there the blaise rest api is not available', () => {
  beforeEach(() => {
    getCaseFactsheetMock.mockRejectedValue(new Error('try again in a few minutes'));
  });

  afterEach(() => {
    getCaseFactsheetMock.mockReset();
  });

  it('should display an error message telling the user to try again in a few minutes', async () => {
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseFactsheet />
        </BrowserRouter>,
      );
    });

    // assert
    const factsheetView = view.getByTestId('Factsheet');
    expect(factsheetView).toHaveTextContent('try again in a few minutes');
  });

  it('should render the page correctly when an error occurs', async () => {
    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <CaseFactsheet />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });
});
