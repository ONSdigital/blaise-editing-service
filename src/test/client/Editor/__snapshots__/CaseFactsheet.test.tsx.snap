// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given there is a case available in blaise for a questionnaire should display the factsheet correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="Factsheet"
      >
        <div
          class="ons-panel ons-panel--info ons-panel--no-title ons-panel--spacious ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            This factsheet contains information for case
             
            1
             
            completed by
             
            Rich
          </div>
        </div>
        <dl
          class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
          style="margin: 2rem 0px 0px 0px;"
          title="Factsheet"
        >
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            CaseId:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            1
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Address:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Flat 1
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Richmond House
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Rice Road
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Duffrin
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Newport
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Gwent
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            NZ11 4PD
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Interviewer:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Rich
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Outcome code:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            110
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Household size:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            2
          </dd>
          <dt
            class="ons-metadata__term ons-grid__col ons-col-3@m"
          >
            Respondents:
          </dt>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Richmond Ricecake
             
            Fri Feb 15 1980 00:00:00 GMT+0000 (Greenwich Mean Time)
          </dd>
          <dd
            class="ons-metadata__value ons-grid__col ons-col-9@m"
          >
            Bartholomew Edgar
             
            Sun Jun 11 1995 00:00:00 GMT+0100 (British Summer Time)
          </dd>
        </dl>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="Factsheet"
    >
      <div
        class="ons-panel ons-panel--info ons-panel--no-title ons-panel--spacious ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          This factsheet contains information for case
           
          1
           
          completed by
           
          Rich
        </div>
      </div>
      <dl
        class="ons-metadata ons-metadata__list ons-grid ons-grid--gutterless ons-u-cf ons-u-mb-no"
        style="margin: 2rem 0px 0px 0px;"
        title="Factsheet"
      >
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          CaseId:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          1
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Address:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Flat 1
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Richmond House
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Rice Road
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Duffrin
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Newport
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Gwent
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          NZ11 4PD
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Interviewer:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Rich
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Outcome code:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          110
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Household size:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          2
        </dd>
        <dt
          class="ons-metadata__term ons-grid__col ons-col-3@m"
        >
          Respondents:
        </dt>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Richmond Ricecake
           
          Fri Feb 15 1980 00:00:00 GMT+0000 (Greenwich Mean Time)
        </dd>
        <dd
          class="ons-metadata__value ons-grid__col ons-col-9@m"
        >
          Bartholomew Edgar
           
          Sun Jun 11 1995 00:00:00 GMT+0100 (British Summer Time)
        </dd>
      </dl>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Given there the blaise rest api is not available should render the page correctly when an error occurs 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="Factsheet"
      >
        <div
          class="ons-panel ons-panel--error ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            try again in a few minutes
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="Factsheet"
    >
      <div
        class="ons-panel ons-panel--error ons-panel--no-title  ons-u-mt-m"
      >
        <div
          class="ons-panel__body "
        >
          try again in a few minutes
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
